### PR TITLE
INT-2212: Emit the number of claimed/unclaimed locks as metrics.

### PR DESCRIPTION
**Pull request by cowdude**
Also fixed counterfeiter usage in Dockerfile...

### CODE REVIEW

File: cmd/out/main.go [hunk 1]
```diff
@@ -129,9 +141,24 @@ func main() {
 		},
 	})
 
+	if err != nil {
```
Comment (marco-m) [inline cmd/out/main.go:144]: it looks like this section has the wrong indentation ?

File: out/outfakes/fake_lock_handler.go [hunk 2]
```diff
@@ -1,45 +1,19 @@
 // Code generated by counterfeiter. DO NOT EDIT.
-package fakes
+package outfakes
 
 import (
-	"sync"
+	sync "sync"
```
Comment (marco-m) [inline out/outfakes/fake_lock_handler.go:5]: Is this change (line 5) needed ?

File: out/outfakes/fake_lock_handler.go [hunk 3]
```diff
@@ -1,45 +1,19 @@
...
 
-	"github.com/concourse/pool-resource/out"
+	out "github.com/concourse/pool-resource/out"
```
Comment (marco-m) [inline out/outfakes/fake_lock_handler.go:7]: Is this change needed?

File: out/outfakes/fake_lock_handler.go (hunk 2 above)
Comment (marco-m) [reply out/outfakes/fake_lock_handler.go:5]: Ah, this file is generated by counterfeiter ?

Comment (marco-m) [reply out/outfakes/fake_lock_handler.go:5]: Is the rename of package needed ? We should try to keep the changes with upstream as few as possible

File: out/outfakes/fake_lock_handler.go [hunk 4]
```diff
@@ -1,45 +1,19 @@
...
-		result2 error
-	}
-	AddLockStub        func(lock string, contents []byte, initiallyClaimed bool) (version string, err error)
+	AddLockStub        func(string, []byte, bool) (string, error)
 	addLockMutex       sync.RWMutex
 	addLockArgsForCall []struct {
-		lock             string
-		contents         []byte
-		initiallyClaimed bool
+		arg1 string
```
Comment (marco-m) [inline out/outfakes/fake_lock_handler.go:14]: the fields names changes here are done by counterfeiter ?

File: out/outfakes/fake_lock_handler.go (hunk 2 above)
Comment (cowdude) [reply out/outfakes/fake_lock_handler.go:5]: Counterfeiter, yes.

File: out/outfakes/fake_lock_handler.go (hunk 3 above)
Comment (cowdude) [reply out/outfakes/fake_lock_handler.go:7]: Counterfeiter.

File: out/outfakes/fake_lock_handler.go (hunk 4 above)
Comment (cowdude) [reply out/outfakes/fake_lock_handler.go:14]: Yes...

File: cmd/out/main.go (hunk 1 above)
Comment (cowdude) [reply cmd/out/main.go:144]: Looks like bitbucket isn't the only one having issues...
https://github.com/Pix4D/pool-resource/blob/6a9dfd677bad5d6c3b121700bc9b79fe3f000a2e/cmd/out/main.go

**Review by aledegano — APPROVED** [review]

**Review by odormond — APPROVED** [review]
